pymc/distributions/distribution.py

```diff
@@ -542,15 +546,45 @@ def dist(
         ndim_supp = cls.ndim_supp(*dist_params)
         create_size = find_size(shape=shape, size=size, ndim_supp=ndim_supp)
         rv_out = cls.rv_op(*dist_params, size=create_size, **kwargs)
-        # This is needed for resizing from dims in `__new__`
-        rv_out.tag.ndim_supp = ndim_supp
 
         rv_out.logp = _make_nice_attr_error("rv.logp(x)", "pm.logp(rv, x)")
         rv_out.logcdf = _make_nice_attr_error("rv.logcdf(x)", "pm.logcdf(rv, x)")
         rv_out.random = _make_nice_attr_error("rv.random()", "pm.draw(rv)")
         return rv_out
 
 
+class SymbolicRandomVariable(OpFromGraph):
+    """TODO"""
+
+    ndim_supp = None
+
+    def __init__(self, *args, ndim_supp, **kwargs):
+        self.ndim_supp = ndim_supp
+        kwargs.setdefault("inline", True)
```

Comment:
What happens if inline is false?

Reply:
When inline is False the inner code of an OpFromGraph remains as a black-box, so compilation does not mix (or optimize) across the outer and inner nodes. This only matters for the generative/random functions, because in the logp functions, the whole node is replaced by the logp expression instead.